ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -883,20 +1036,43 @@ export const RuntimePanel = fp.flow(
   if (currentRuntime === undefined) {
     return <Spinner style={{width: '100%', marginTop: '5rem'}}/>;
   }
-
+  const createStandardComputeRuntimeRequest = (runtime: RuntimeConfig) => {
+    // The logic here is tricky to be compatible
+    // Use gceConfig when (PD feature is not enabled) OR (increase PD size or update machineType of an existing runtime with PD attached)
+    // OR (update an existing runtime with no PD attached).
+    // TODO(RW-): 'Update an existing runtime with no PD attached' will be cleaned up
+    // post launch PD when all existing running Runtime shutdown.
+    if (!enablePD || (!pdSizeReduced && runtimeExists) || runtimeExistsWithoutPD) {
+      return {
+        gceConfig: {
+          machineType: runtime.machine.name,
+          diskSize: !pdExists ? runtime.diskSize : runtime.pdSize
+        }
+      };
+    } else {
+      return {
+        gceWithPdConfig: {
+          bootDiskSize: 50,
```

Comment:
I'm pretty sure this is still ignored in both cases - it should also not be included her if we're omitting from the normal gceConfig.

Reply:
not addressed

Reply:
@YintongMa per CH's question, is bootDiskSize used in PDConfig. It is confusing that we have `bootDiskSize` and  a size in `persistentDisk` block